redwood-treehouse-host/src/iosMain/kotlin/app/cash/redwood/treehouse/TreehouseUIView.kt

```diff
@@ -50,26 +54,41 @@ public class TreehouseUIView private constructor(
   public constructor(widgetSystem: WidgetSystem<UIView>) : this(widgetSystem, RootUiView())
 
   init {
-    (view as RootUiView).treehouseView = this
+    view.treehouseView = this
   }
 
   private fun superviewChanged() {
     readyForContentChangeListener?.onReadyForContentChanged(this)
   }
 
-  private class RootUiView : UIView(cValue { CGRectZero }) {
+  /**
+   * The root view is just a vertical stack.
+   *
+   * In practice we expect this to contain either zero child subviews (especially when
+   * newly-initialized) or one child subview, which will usually be a layout container.
+   *
+   * This could just as easily be a horizontal stack. A box would be even better, but there's no
+   * such built-in component and implementing it manually is difficult if we want to react to
+   * content resizes.
```

Comment:
Isn't the behavior of the root view the same that we want out of a UIKit `Box` which we also are required to provide though? Does that mean our UIKit `Box` is broken when its set to wrap/wrap and one of its children changes size?

Reply:
Yep, it’s quite likely Box is broken in the same way that TreehouseUIView is broken.

Perhaps worth doing the hacky work to observe child size changes for that... sigh.